torch_geometric/transforms/mask.py

```diff
@@ -0,0 +1,88 @@
+from typing import List, Optional, Union
+
+from torch_geometric.data import Data
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+from torch_geometric.utils import index_to_mask, mask_to_index
+
+
+def attrs_ends_with(data: Data, suffix: str):
+    return tuple(filter(lambda s: s.endswith(suffix), data.keys))
+
+
+def update_suffix(name, old_suffix, new_suffix):
+    if name.endswith(old_suffix):
+        name = name[:-len(old_suffix)]
+
+    return name + new_suffix
+
+
+@functional_transform('index_to_mask')
+class IndexToMask(BaseTransform):
+    r"""Converts indices to a mask representation
+
+    Args:
+        attrs (str, [str], optional): If given, will only perform index to mask
+            conversion for the given attributes. If omitted, will infer the
+            attributes from the suffix `_indices`  (default: :obj:`None`)
+        sizes (int, [int], optional). The size of the mask. If set to
+            :obj:`None`, a minimal sized mask is created.
+            (default: :obj:`None`)
+    """
+    def __init__(self, attrs: Optional[Union[str, List[str]]] = None,
+                 sizes: Optional[Union[int, List[int]]] = None):
+        self.attrs = (attrs, ) if isinstance(attrs, str) else attrs
+        self.sizes = sizes
+
+    def __call__(self, data: Data) -> Data:
```

Comment:
Can we let this support both `data` and `HeteroData`? We can simply iterate over all `node_stores` and `edge_stores` to support this.

Reply:
thanks I wasn't sure how to do this with `HeteroData` - agree it would be better to support this.

Reply:
I think this is working now for `HeteroData` but still wanted to add a test.  Let me know if you spot anything obviously missing in the implementation while I work on that.